File: algorithms/1_easy/B/01920_build_array_from_permutation/solution.cpp
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        const size_t len{nums.size()};
        vector<int> res(len, 0);

        for (size_t i{0}; i < len; ++i) {
            res[i] = nums[static_cast<size_t>(nums[i])];
        }

        return res;
    }
};

File: algorithms/1_easy/B/01920_build_array_from_permutation/solution.rs
impl Solution {
    pub fn build_array(nums: Vec<i32>) -> Vec<i32> {
        let mut res: Vec<i32> = vec![0; nums.len()];

        for i in 0..nums.len() {
            res[i] = nums[nums[i] as usize];
        }

        res
    }
}
